src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -36,6 +36,9 @@
 #include "DataStructures/Tensor/IndexType.hpp"
 #include "DataStructures/Tensor/Structure.hpp"
 #include "DataStructures/Tensor/TypeAliases.hpp"
+#ifdef SPECTRE_AUTODIFF
```

Comment:
This doesn't scale. You should have the `ifdef` inside the `Autodiff/Autodiff.hpp` header. We can't have `ifdefs` everywhere we include a file.